playback/src/audio_backend/pulseaudio.rs

```diff
@@ -47,13 +45,18 @@ impl From<PulseError> for SinkError {
 }
 
 pub struct PulseAudioSink {
-    s: Option<Simple>,
+    sink: Option<Simple>,
     device: Option<String>,
+    app_name: String,
+    stream_desc: String,
     format: AudioFormat,
 }
 
 impl Open for PulseAudioSink {
     fn open(device: Option<String>, format: AudioFormat) -> Self {
+        let app_name = env::var("PULSE_PROP_application.name").unwrap_or_default();
```

Comment:
This results in an empty name if this crate is used as library and the user doesn't set `"PULSE_PROP_application.name"`. This manifests itself for example in [`ncspot`](github.com/hrkfdn/ncspot) where there is no explicit assignment to any `"PULSE_PROP_application"` environment variable.

![2022-07-20-181107](https://user-images.githubusercontent.com/9497885/180033521-305e3245-aced-4e70-9fd5-f98d7967159c.png)


Previously this was unconditionally:

```rust
const APP_NAME: &str = "librespot";
```

Should the environment variable settings be also done here, in the library version, rather than in `main.rs`?

Or at least have `"librespot"` as the fallback, instead of an empty string:

```rust
let app_name = env::var("PULSE_PROP_application.name").unwrap_or("librespot");
```

Likewise for `stream_desc` on the following line.

Reply:
No, the default is an empty string on purpose. It's up to the consuming program or user to set the prop.

Reply:
Librespot's main is not a part of librespot the set of libs. A consumer of the libs would have their own main where they can set the var to whatever they like.

Reply:
This was clearly listed as a breaking change if you have an issue with ncspot file a bug report with them.

Reply:
Librespot the lib should not impose it's name on consuming programs.

Reply:
It's blank because a consuming program should set it.

Reply:
The props are completely optional. Not being set doesn't compromise functionality.

Reply:
Should not must.